packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -119,30 +146,50 @@ const hasValidParticipant = (chatMessage: ChatMessageWithStatus): boolean =>
  * @public
  */
 export const messageThreadSelector: MessageThreadSelector = createSelector(
-  [getUserId, getChatMessages, getLatestReadTime, getIsLargeGroup],
-  (userId, chatMessages, latestReadTime, isLargeGroup) => {
+  [getUserId, getChatMessages, getLatestReadTime, getIsLargeGroup, getReadInformation, getParticipants],
+  (userId, chatMessages, latestReadTime, isLargeGroup, readInformation, participants) => {
+    // get number of participants
+    const messageThreadParticipantCount = Object.values(participants).length - 1;
+    // for each chat message, add read receipt information when readInformation.id === chatmessage.id
+    const convertedChatMessages = Object.values(chatMessages).map(function (message) {
+      const readReceipt = Object.values(readInformation).filter((info) => info.chatMessageId === message.id);
+      // readReceipt is getting duplicate information, remove duplicate here
+      const uniqueReadReceipt = readReceipt.filter(
```

Comment:
Comment for following work items after this PR:
We might need to implement readReceipt logic here for readReceipt contextual menu - once we detect one message has been read by someone - then we assume that person read all the message above
Which means we might wanna change data structure a little bit for the next version

we might need a lastReadLine or lastReadMessage for each person, when we render readReceipt just follow the assumption we made

This is just a thought (and notes) for next work item, let's have more discussion after we finish this one

Reply:
Updated PR with new logic